src/topology/uniform_space/cauchy.lean

```diff
@@ -712,3 +718,35 @@ begin
 end
 
 end uniform_space
+
+section covariant
+
+/-- Proves things like `cauchy_seq f → cauchy_seq (λ (n : ℕ), f (1+n))` by `library_search`. -/
+lemma cauchy_seq.covariant_of_const_left_nat {f : ℕ → α} {μ : ℕ → ℕ → ℕ}
+  [covariant_class ℕ ℕ μ (≤)] (hf : cauchy_seq f) {m : ℕ} : cauchy_seq (λ n, f (μ m n)) :=
```

Comment:
I feel that this kind of lemma would be much more natural without typeclass assumptions, i.e., in the form, if `u` is monotone and `f` is a Cauchy sequence, then `λ n, f (u n)` is also a Cauchy sequence. Same thing for the next lemmas.

Reply:
I don't think your version could be found by `library_search`, which was my main motivation for this PR. Another option would be to prove it directly for `λ n, f (n+c)` and `λ n, f (c+n)`, which would accomplish the goal for all the cases I encountered while writing #11908 as well as the one in #12215. This would allow me to remove `monotone.tendsto_at_top_at_top_or_eventually_const_*` lemmas, since the `eventually_const` case is not relevant for addition. Would this be preferable?

As someone who's new to Mathlib, I rely on `library_search` a lot to find these sorts of helper lemmas. That's why this kind of change is important to me.

Reply:
I don't think that having lemmas that can be found by `library_search` should be the main objective here: there are other ways to find relevant lemmas, and hopefully much better tools than `library_search` are coming. For instance, if I wanted to find lemmas saying that Cauchy sequences are preserved by composition with monotone maps, I would grep the library for `cauchy_seq.*monotone`. So if you make sure that your lemma names correspond to this pattern, they would be easy to find. I'd name them `cauchy_seq.of_monotone`, for the record.